WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.kt

```diff
@@ -0,0 +1,353 @@
+        FILTER_ALL,
+        FILTER_COMMENT,
+        FILTER_FOLLOW,
+        FILTER_LIKE,
+        FILTER_UNREAD;
+
+        override fun toString(): String {
+            return when (this) {
+                FILTER_ALL -> "all"
+                FILTER_COMMENT -> "comment"
+                FILTER_FOLLOW -> "follow"
+                FILTER_LIKE -> "like"
+                FILTER_UNREAD -> "unread"
+            }
+        }
+    }
+
+    var currentFilter = FILTERS.FILTER_ALL
+        private set
+    private var mReloadNotesFromDBTask: ReloadNotesFromDBTask? = null
+
+    interface DataLoadedListener {
+        fun onDataLoaded(itemsCount: Int)
+    }
+
+    interface OnLoadMoreListener {
+        fun onLoadMore(timestamp: Long)
+    }
+
+    private var mOnNoteClickListener: OnNoteClickListener? = null
+    fun setFilter(newFilter: FILTERS) {
+        currentFilter = newFilter
+    }
+
+    fun addAll(notes: List<Note>, clearBeforeAdding: Boolean) {
+        notes.sortedWith(TimeStampComparator())
+        try {
+            if (clearBeforeAdding) {
+                mNotes.clear()
+            }
+            mNotes.addAll(notes)
+        } finally {
+            myNotifyDatasetChanged()
+        }
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private fun myNotifyDatasetChanged() {
+        buildFilteredNotesList(mFilteredNotes, mNotes, currentFilter)
+        notifyDataSetChanged()
+        mDataLoadedListener.onDataLoaded(itemCount)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.notifications_list_item, parent, false)
+        return NoteViewHolder(view)
+    }
+
+    private fun getNoteAtPosition(position: Int): Note? {
+        return if (isValidPosition(position)) {
+            mFilteredNotes[position]
+        } else null
+    }
+
+    private fun isValidPosition(position: Int): Boolean {
+        return position >= 0 && position < mFilteredNotes.size
+    }
+
+    override fun getItemCount(): Int {
+        return mFilteredNotes.size
+    }
+
+    @Suppress("CyclomaticComplexMethod", "LongMethod")
+    override fun onBindViewHolder(noteViewHolder: NoteViewHolder, position: Int) {
+        val note = getNoteAtPosition(position) ?: return
+        noteViewHolder.mContentView.tag = note.id
+
+        // Display group header
+        val timeGroup = Note.getTimeGroupForTimestamp(note.timestamp)
+        var previousTimeGroup: NoteTimeGroup? = null
+        if (position > 0) {
+            val previousNote = getNoteAtPosition(position - 1)
+            previousTimeGroup = Note.getTimeGroupForTimestamp(
+                previousNote!!.timestamp
+            )
+        }
+        if (previousTimeGroup != null && previousTimeGroup == timeGroup) {
+            noteViewHolder.mHeaderText.visibility = View.GONE
+        } else {
+            noteViewHolder.mHeaderText.visibility = View.VISIBLE
```

Comment:
These lines can be refactored as following code snippet:
```
noteViewHolder.mHeaderText.isGone = previousTimeGroup != null && previousTimeGroup == timeGroup
```

Reply:
For this one, I simplified it, but I still kept the enclosing else to match the existing logic (i.e. the `when` clause that follows).